### ISSUE TITLE
Failure if foreign make tries to do a rename()

### ISSUE THREAD

**Issue by vrusinov** [bazelbuild/rules_foreign_cc#361]
I had an obscure problem when make would fail with some generic error. I've managed to debug it into following:

One of the tools that my Makefile called had an assumption that all of build root would be on the same filesystem. So then it tried to do a `rename(".tmpconfig.h", "include/generated/autoconf.h") = -1 EXDEV (Invalid cross-device link)` which as you can see did not work.

Here's why:

BUILD_TMPDIR is in /tmp on my machine which is mounted as tmpfs.
But the 'include' directory is symlinked from `~/.cache/bazel/<...>`:
```
$ cd $BUILD_TMPDIR
$ ls -l
lrwxrwxrwx 1 xxx xxx 132 Jan 15 23:34 include -> /home/<...>/.cache/bazel/<...>
l
```

It would be nice to have an option to just copy everything instead of symlinking. Or, perhaps, not use `mkdtmp` and create BUILD_TMPDIR somewhere inside bazel cache directory.

**Comment by vrusinov** [comment]
I think https://github.com/bazelbuild/rules_foreign_cc/pull/367 may fix this

**Comment by jsharpe** [comment]
This should be fixed by #479.
